python/lsst/daf/base/baseLib.i

```diff
@@ -161,6 +161,48 @@ def _propertyContainerGet(container, name, asArray=False):
         pass
     raise lsst.pex.exceptions.TypeError('Unknown PropertySet value type for ' + name)
 
+import numbers
+import sys
+def _guessIntegerType(container, name, value):
+    """Given an existing container and name, determine the type
+    that should be used for the supplied value. The supplied value
+    is assumed to be a scalar.
+
+    On Python 3 all ints are LongLong but we need to be able to store them
+    in Int containers if that is what is being used (testing for truncation).
+    Python3 uses sys.maxsize as an equivalence for the cross over from int to long.
+
+    If there is no pre-existing value we have to decide what to do. For now
+    we pick Int if the value is less than maxsize.
+
+    On Legacy Python always returns None.
+
+    Returns None if the value supplied is not an integral value."""
+    useType = None
+
+    # Should we do this? It will possibly lead to different behavior
+    # between Python versions where passing in long(3) in Legacy Python
+    # will be different to passing in "long(3)" on Python 3.
+    if sys.version_info[0] < 3:
+        return useType
+
+    if isinstance(value, numbers.Integral):
+        try:
+            containerType = _propertyContainerElementTypeName(container, name)
+            if containerType == "Int":
+                if value <= sys.maxsize:
+                    useType = "Int"
+            elif containerType == "LongLong":
+                useType = "LongLong"
```

Comment:
Put the `if.. elif` block in the `try`'s `else` block?  You're not capturing an exception from anything except the `containerType` line.